src/compat/posix/time/strptime.c

```diff
@@ -0,0 +1,345 @@
+			}
+			tm->tm_hour = date;
+			return &new_s[n_leng];
+
+		case 'M':
+			if ((date > 59)) {
+				return NULL;
+			}
+			tm->tm_min = date;
+			return &new_s[n_leng];
+
+		case 'S':
+			if ((date > 60)) {
+				return NULL;
+			}
+			tm->tm_sec = date;
+			return &new_s[n_leng];
+
+		case 'j':
+			if ((date > 366) || (date == 0)) {
+				return NULL;
+			}
+			tm->tm_yday = date - 1;
+			return &new_s[n_leng];
+
+		case 'd':
+		case 'e':
+			if ((date > 31) || (date == 0)) {
+				return NULL;
+			}
+			tm->tm_mday = date;
+			return &new_s[n_leng];
+
+		case 'C':
+			if ((date > 99)) {
+				return NULL;
+			}
+			tm->tm_year = date * 100 - 1900;
+			return &new_s[n_leng];
+
+		case 'U':
+		case 'W':
+			if ((date > 53)) {
+				return NULL;
+			}
+			return &new_s[n_leng];
+
+		case 'n':
+		case 't':
+			if (new_s[0] == ' ') {
+				return &new_s[1];
+			}
+			return &new_s[0];
+
+		case 'R':
+		case 'T':
+			res_equi = def_format("H", &new_s[0], tm);
+			if (res_equi == NULL) {
+				return NULL;
+			}
+			while (new_s[i] != res_equi[0]) {
+				i++;
+			}
+			if (new_s[i] != ':' ) {
+				return NULL;
+			}
+			res_equi = 0;
+			res_equi = def_format("M", &new_s[i + 1], tm);
+			if (res_equi == NULL) {
+				return NULL;
+			}
+			i++;
+			while (new_s[i] != res_equi[0]) {
+				i++;
+			}
+			if (new_form[0] == 'R') {
+				return &new_s[i];
+			}
+			if (new_s[i] != ':' ) {
+				return NULL;
+			}
+			res_equi = 0;
+			res_equi = def_format("S", &new_s[i + 1], tm);
+			if (res_equi == NULL) {
+				return NULL;
+			}
+			while (new_s[i] != res_equi[0]) {
+				i++;
+			}
+			return &new_s[i];
+
+		case 'D':
+			res_equi = def_format("m", &new_s[0], tm);
+			if (res_equi == NULL) {
+				return NULL;
+			}
+			while (new_s[i] != res_equi[0]) {
+				i++;
+			}
+			if (new_s[i] != '/' ) {
+				return NULL;
+			}
+			res_equi = 0;
+			res_equi = def_format("d", &new_s[i + 1], tm);
+			if (res_equi == NULL) {
+				return NULL;
+			}
+			i++;
+			while (new_s[i] != res_equi[0]) {
+				i++;
+			}
+			if (new_s[i] != '/' ) {
+				return NULL;
+			}
+			res_equi = 0;
+			res_equi = def_format("y", &new_s[i + 1], tm);
+			if (res_equi == NULL) {
+				return NULL;
+			}
+			while (new_s[i] != res_equi[0]) {
+				i++;
+			}
+			return &new_s[i];
+
+		default:
+			return NULL;
+	}
+}
+
+static char *new_s_low(char *new_s) {
+	int count = 0;
+	while ((new_s[count] >= 'A') && (new_s[count] <= 'Z')) {
+		new_s[count] = tolower(new_s[count]);
+		count++;
+	}
+	return &new_s[0];
+}
+
+/**
+ * The strptime implementation doesn't support such format characters as %c, %p, %r, %U, %W, %x, %X.
+ * Also it doesn't support all alternative locale-dependent versions of date and time representation
+ * and all format characters of glibc.
+ */
+
+char *strptime(const char *s, const char *format, struct tm *tm) {
+
+	int count_s = 0;
+	int count_f = 0;
+	char *except_percent, *res;
+	int number_s = strlen(s);
+	int number_f = strlen(format);
+	char new_form[number_f + 1];
```

Comment:
Is it possible to avoid allocating new arrays? Strings may be pretty long, so it's better not to use stack for such arrays.

Reply:
Yes, I changed it.
Thank you!